### PR TITLE
Added gzip support

### PR DESCRIPTION
**Pull request by m1ome**
Hi, awesome tool, but we lacking of support `gzip`.

Implemented it.

### CODE REVIEW

**Comment by m1ome** [comment]
@gchaincl changed PR

File: dump.go [hunk 1]
```diff
@@ -21,9 +22,30 @@ func withColor(color int, text string) string {
...
 
 func writeBody(buf *bytes.Buffer, req *http.Request) error {
-	body, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		return err
+	var body []byte
+	var err error
+
+	if strings.Contains(req.Header.Get("Accept-Encoding"), "gzip") {
+		gz, err := gzip.NewReader(req.Body)
```
Comment (gchaincl) [inline dump.go]: I'm still getting errors, I think this happens when Browser send the Headers `Accept-Encoding: gzip` but the content is not gzipped.

In the attached screen shot you can see that the Info bar is displaying an EOF
2017-04-10-095833_1599x830_scrot

I think that case should be handled as well.

Comment (m1ome) [inline dump.go]: @gchaincl i will try to reproduce it, but how should it be handled?

Comment (gchaincl) [inline dump.go]: To reproduce it, I just made a request with the browser.
I think we should fallback to the original behavior, which is display the body as is.

Comment (m1ome) [inline dump.go]: @gchaincl i will try to avoid additional body read with this behaviour, give me a lil time for it.

Comment (gchaincl) [inline dump.go]: sure, no hurries

**Comment by m1ome** [comment]
@gchaincl added checking + added some test for correct stuff look

**Comment by gchaincl** [comment]
@m1ome after you opened this pr I've been reading about http compression, and it looks like the header `Accept-Encoding:gzip`  is set by the client to specify that he is able to decode gzipped responses, but not that the request is compressed [1].
Also compression occurs after a negotiation process, which means that both parties should _agree_, this means that if `HTTPLab` doesn't `Accept-Encoding:gzip` the client should never send gzipped data, and if he does, it should be considered a bug.

I might be wrong, so please double check the protocol docs and verify if the behavior that you're implementing is right or not.

Thanks and sorry for the delay!

[1]: https://en.wikipedia.org/wiki/HTTP_compression#Compression_scheme_negotiation

**Comment by gchaincl** [comment]
@m1ome, thoughts on this?

**Comment by m1ome** [comment]
@gchaincl not yet, just have a lot non OSS work. Trying to get things done on this week.

**Comment by gchaincl** [comment]
:1: let me know if I can provide any help

**Comment by m1ome** [comment]
@gchaincl updated PR

**Comment by gchaincl** [comment]
@m1ome this looks much better, thanks! Mind to fix conflicts before it can be merged?

**Comment by m1ome** [comment]
@gchaincl i will rebase master on this

**Comment by gchaincl** [comment]
@m1ome is this still active?

**Comment by m1ome** [comment]
@gchaincl rebased master and then nothing heard from you :)

**Comment by gchaincl** [comment]
Apparently, it still has conflicts.

**Comment by m1ome** [comment]
Strange I see:
**This branch has no conflicts with the base branch**

**Comment by joequery** [comment]
@gchaincl what branch are you trying to pull this into? The PR is set to merge into `master` but it may have conflicts with the `cors` branch you are working on.

**Comment by m1ome** [comment]
@joequery CORS branch?

**Comment by gchaincl** [comment]
@m1ome this is what I mean:
branch

**Comment by gchaincl** [comment]
@joequery my guess is that GH do this for master branch

**Comment by m1ome** [comment]
@gchaincl why do you wanna rebase and merge? Seems strange you can do Squash + Merge. Of course you cannot rebase due to merged master in this branch and this is conflicts.

**Comment by gchaincl** [comment]
I just need to resolve conflicts, otherwise, it can't be merged

**Comment by joequery** [comment]
@m1ome I'm referring to [this branch](https://github.com/gchaincl/httplab/tree/cors) which is a little ahead of master right now. I was wondering if @gchaincl was attempting to pull into the wrong branch.

@gchaincl The reason why we are confused is because this is what we see

screen shot 2017-07-03 at 9 48 29 am

**Comment by gchaincl** [comment]
@joequery that screenshot if from CORS branch?

**Comment by joequery** [comment]
@gchaincl No, from this PR

**Comment by gchaincl** [comment]
so @joequery what's the thing about CORS branch then

**Comment by joequery** [comment]
I was guessing you might have been trying to pull into the wrong branch. Because there's no merge conflict here.

**Comment by gchaincl** [comment]
Well, that's weird, so according to you, this PR has no conflict?
Can I see a bigger ss please, now I'm scared!

**Comment by joequery** [comment]
Unfortunately the comment thread is too long for a screenshot :P If you aren't trusting the UI then you can create your own branch off master and merge these changes into it and push up your own PR.

**Comment by gchaincl** [comment]
It's not about trusting or not, I think there's a problem and I want to find out what's happening.

**Comment by gchaincl** [comment]
It's not about trusting or not, I think there's a problem and I want to find out what's happening.
branch

**Comment by gchaincl** [comment]
try this: update your fork's master branch with origin, and then try to rebase from the CLI

**Comment by gchaincl** [comment]
I'm closing this as seems abandoned, but feel free to re-open if you continue working on it

**Pull request closed without merging**